static/src/javascripts/projects/commercial/modules/dfp/prepare-permutive.js

```diff
@@ -0,0 +1,137 @@
+                            const o = Array.prototype.slice.call(arguments, 0);
+                            e.q.push({ functionName: n, arguments: o });
+                        };
+                    })(f);
+                }
+            }
+        })(
+            document,
+            window.permutive,
+            'd6691a17-6fdb-4d26-85d6-b3dd27f55f08',
+            '359ba275-5edd-4756-84f8-21a24369ce0b',
+            {}
+        );
+        (window.googletag = window.googletag || {}),
+            (window.googletag.cmd = window.googletag.cmd || []),
+            window.googletag.cmd.push(() => {
+                if (
+                    window.googletag.pubads().getTargeting('permutive')
+                        .length === 0
+                ) {
+                    const g = window.localStorage.getItem('_pdfps');
+                    window.googletag
+                        .pubads()
+                        .setTargeting('permutive', g ? JSON.parse(g) : []);
+                }
+            });
+        /* eslint-enable */
+        try {
+            const {
+                isPaidContent,
+                pageId,
+                headline,
+                contentType,
+                section,
+                author,
+                keywords,
+                webPublicationDate,
+                series,
+            } = config.get('page');
+
+            const safeAuthors =
+                author && typeof author === 'string' ? author.split(',') : null;
+            const safeKeywords =
+                keywords && typeof keywords === 'string'
+                    ? keywords.split(',')
+                    : null;
+            const safePublishedAt =
+                webPublicationDate && typeof webPublicationDate === 'number'
+                    ? new Date(webPublicationDate).toISOString()
+                    : null;
+            const rawPayload = {
+                premium: isPaidContent,
+                id: pageId,
+                title: headline,
+                type: contentType,
+                section,
+                authors: safeAuthors,
+                keywords: safeKeywords,
+                publishedAt: safePublishedAt,
+                series,
+            };
+
+            const isEmpty = value =>
+                value === '' || typeof value === 'undefined' || value === null;
+
+            const removeEmpty = payload => {
+                Object.keys(payload).forEach(
+                    key => isEmpty(payload[key]) && delete payload[key]
+                );
+                return payload;
+            };
+
+            const payload = removeEmpty(rawPayload);
+            permutive.addon('web', {
+                page: {
+                    content: payload,
+                },
+            });
+        } catch (err) {
+            reportError(err, { feature: 'comercial' }, false);
+        }
+
+        return resolve();
+    });
+};
+
+export const initPermutiveLib = (): Promise<void> => {
```

Comment:
Should this be in a separate file rather than hiding at the bottom of this one? I'm not sure, could be convinced either way. What do you think?

Super minor point :-)

Reply:
I'd argue for keeping these two in one file, as it doesn't offend me that the file is longer and these two parts of the integration won't work without each other. Would a header comment at the top explaining this function duo help?

Reply:
Nah all good then.